### PR TITLE
UI: Improve NumberPad UX

### PR DESCRIPTION
**Pull request by pwltr**
# Description

### Changes
- improve NumberPad UX
- update all NumberPads (except SendNumberPad & NumberPadLightning)
- remove redundant `unitPreference`
- show only 5 decimals for BTC amounts in activity
- fix TextField font weight
- add e2e tests for ChangePIN
- add e2e tests for Receive

### Notes
Rewrite for SendNumberPad & NumberPadLightning will come in another PR. Dealing with the edge cases there would've blown up the scope.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Tests
- [x] Detox test
- [x] Unit test

## Screenshot / Video

https://user-images.githubusercontent.com/8538369/231811417-8bb4b980-7773-4a60-a55a-e64cd7063584.mp4

## QA Notes

Will have 100% Coverage by automated tests

### CODE REVIEW

File: src/components/MoneySymbol.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+}: {
+	unit: EBalanceUnit;
+	style?: StyleProp<ViewStyle>;
+}): ReactElement => {
+	const { fiatSymbol } = useCurrency();
+
+	return (
+		<View style={style}>
+			{unit === EBalanceUnit.fiat && (
+				<Display color="white5" money={true} testID="MoneyFiatSymbol">
```
Comment (limpbrains) [inline src/components/MoneySymbol.tsx]: I removed `money` from Display props

File: __tests__/numberpad.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
+import { updateUi } from '../src/store/actions/ui';
+import { handleNumberPadPress } from '../src/utils/numberpad';
+
+// Fix 'getDispatch is not a function'
+updateUi;
```
Comment (limpbrains) [inline __tests__/numberpad.ts]: I fixed same issue by adding 
`import '../src/utils/i18n';` to the beginning of the test

File: src/components/NumberPad.tsx [hunk 3]
```diff
@@ -6,146 +6,105 @@ import { BackspaceIcon } from '../styles/icons';
...
-		);
-	},
+	}): ReactElement => (
+		<TouchableOpacity
+			style={styles.buttonContainer}
+			color="transparent"
+			activeOpacity={ACTIVE_OPACITY}
+			testID={testID}
+			onPress={onPress}>
+			<Text style={[styles.button, hasError && { color: '#ff6600' }]}>
```
Comment (limpbrains) [inline src/components/NumberPad.tsx]: inline style

**Review by limpbrains — CHANGES_REQUESTED** [review]
Why on Send screen I don't have this nice grey hightlightning for Amount input? It is hard to enter amount in BTC.

Amazing work, especially all the tests !

**Comment by pwltr** [comment]
> Why on Send screen I don't have this nice grey hightlightning for Amount input? It is hard to enter amount in BTC.

Thanks, for the Send and Lightning onboarding NumberPads I will open a separate PR. There are a lot of edge cases to handle so I split the work up a bit.

**Review by limpbrains — APPROVED** [review]
